src/gsy_e/gsy_e_core/util.py

```diff
@@ -284,12 +295,13 @@ def convert_nested_settings(class_object, class_name, settings_dict):
                 continue
             convert_nested_settings(settings_class, settings_class_name, const_settings)
         return const_settings
-    except Exception:
+    except Exception as ex:
         raise SyntaxError("Error when serializing the const settings file. Incorrect "
-                          "setting structure.")
+                          "setting structure.") from ex
 
 
 def retry_function(max_retries=3):
+    """Retry function."""
```

Comment:
`Decorator that retries the execution of the function until it returns without raising an exception`